qiskit_experiments/framework/base_experiment.py

```diff
@@ -265,20 +265,73 @@ def _finalize(self):
         """
         pass
 
-    def _run_jobs(self, circuits: List[QuantumCircuit], **run_options) -> List[Job]:
-        """Run circuits on backend as 1 or more jobs."""
+    def _max_circuits(self, backend: Backend = None):
+        """
+        Calculate the maximum number of circuits per job for the experiment.
+        """
+
+        # set backend
+        if backend is None:
+            if self.backend is None:
+                raise QiskitError("A backend must be provided.")
+            backend = self.backend
+        self.backend = backend
```

Comment:
One last thing: I don't think `self.backend` should be set to `backend` here. With the current logic, if `job_info()` provides a new backend, then this backend will overwrite the current backend, which isn't behavior expected from a method that should just provide information. I think the PR can be merged after this line is removed.

Reply:
Hi @coruscating Sorry for the delay
I think this will fix it
```python {
        if backend is None:
            if self.backend is None:
                raise QiskitError("A backend must be provided.")
            backend = self.backend
        # Get max circuits for job splitting
        max_circuits_option = getattr(self.experiment_options, "max_circuits", None)
        max_circuits_backend = backend.max_circuits
```
![image](https://github.com/Qiskit-Extensions/qiskit-experiments/assets/81795258/d68365a0-3760-44fa-b169-a8bd726ad878)

here are the results also I tried on `test_framework.py` run successfully
```
{'Total number of circuits in the experiment': 200, 'Maximum number of circuits per job': 100, 'Total number of jobs': 2}
{'Total number of circuits in the experiment': 200, 'Maximum number of circuits per job': 50, 'Total number of jobs': 4}
{'Total number of circuits in the experiment': 200, 'Maximum number of circuits per job': 100, 'Total number of jobs': 2}
{'Total number of circuits in the experiment': 200, 'Maximum number of circuits per job': 110, 'Total number of jobs': 2}
[<qiskit_experiments.test.utils.FakeJob object at 0x0000025549DA24D0>, <qiskit_experiments.test.utils.FakeJob object at 0x0000025549DA3250>]
```
let me know if need further changes else I push the changes

Reply:
That looks good, thanks!

Reply:
Sorry, there's one more issue. `backend.max_circuits` only works for V2 backends, which is why we have the `BackendData` adaptor class that implements the `max_circuits` property: https://github.com/Qiskit-Extensions/qiskit-experiments/blob/cfb47e24fae4173ea106de4b84e686a7933f8589/qiskit_experiments/framework/backend_data.py#L169-L176

So instead of `max_circuits_backend = backend.max_circuits` you should do `max_circuits_backend = BackendData(backend).max_circuits`